tools/app.hpp

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <thread>
+
+#include <CLI/CLI.hpp>
+#include <range/v3/view/getlines.hpp>
+#include <range/v3/view/transform.hpp>
+
+#include "io.hpp"
+#include "query/queries.hpp"
+
+namespace pisa {
+
+namespace arg {
+
+    struct Index {
+        explicit Index(CLI::App *app)
+        {
+            app->add_option("-i,--index", m_basename, "Inverted index basename")->required();
+            app->add_option("-e,--encoding", m_encoding, "Index encoding")->required();
```

Comment:
I agree that `-e` is better than `-t` but I'm sad that it will break my scripts...

Reply:
Yeah, this is up for debate, but I think it will hurt less now than later if we wanted to change it :) I think we can also simply use both with `"-i,-t,--encoding"`. But I'm not so sure that's a great idea. @amallia thoughts?

Reply:
@JMMackenzie @amallia take a look at the latest commit and tell me what you think.

Reply:
Looks cool, that's one way to fix it in the meantime :-) We can then deprecate the legacy options later? Though I am willing to pay the price mow if we want to just go for the new options.

Reply:
I think the good time would probably be v1.0 to get rid of it. But on the other hand, maybe we want to reuse `-t` for something different, then we might wanna remove it right away. I don't have a clear preference. I think it might come down to @amallia